app/uk/gov/hmrc/tariffclassificationfrontend/models/Permission.scala

```diff
@@ -16,36 +16,232 @@
 
 package uk.gov.hmrc.tariffclassificationfrontend.models
 
-import uk.gov.hmrc.tariffclassificationfrontend.models.Role.Role
+trait Permission {
+  def name: String
+}
+
+trait GlobalPermission extends Permission {
+  def appliesTo(operator: Operator): Boolean
+}
+
+trait CasePermission extends Permission {
+  def appliesTo(`case`: Case, operator: Operator): Boolean
+}
+
+object Permission {
+  private lazy val values: Set[Permission] = Set(
+    VIEW_MY_CASES,
+    VIEW_QUEUE_CASES,
+    VIEW_ASSIGNED_CASES,
+    SEARCH,
+    ADVANCED_SEARCH,
+    VIEW_CASES,
+    VIEW_CASE_ASSIGNEE,
+    VIEW_REPORTS,
+    CREATE_CASES,
+    ASSIGN_CASE,
+    RELEASE_CASE,
+    SUPPRESS_CASE,
+    REFER_CASE,
+    REOPEN_CASE,
+    REJECT_CASE,
+    SUSPEND_CASE,
+    COMPLETE_CASE,
+    REMOVE_ATTACHMENTS,
+    CANCEL_CASE,
+    ADD_NOTE,
+    ADD_ATTACHMENT,
+    KEYWORDS,
+    EDIT_LIABILITY,
+    EDIT_RULING,
+    APPEAL_CASE,
+    EXTENDED_USE,
+    MOVE_CASE_BACK_TO_QUEUE,
+    MANAGE_KEYWORDS,
+    MANAGE_QUEUES,
+    EDIT_SAMPLE
+  )
+
+  def from(string: String): Option[Permission] = values.find(_.name == string)
+  def applyingTo(operator: Operator): Set[Permission] = values
+    .filter(_.isInstanceOf[GlobalPermission])
+    .map(_.asInstanceOf[GlobalPermission])
+    .filter(_.appliesTo(operator))
+    .map(_.asInstanceOf[Permission])
+  def applyingTo(`case`: Case, operator: Operator): Set[Permission] = values
+    .filter(_.isInstanceOf[CasePermission])
+    .map(_.asInstanceOf[CasePermission])
+    .filter(_.appliesTo(`case`, operator))
+    .map(_.asInstanceOf[Permission])
+
+  private def anyone(): Boolean = true
+
+  private def managersOrAssignedTeamMembersOnly(`case`: Case, operator: Operator): Boolean = operator.role match {
+    case Role.CLASSIFICATION_MANAGER => true
+    case Role.CLASSIFICATION_OFFICER if `case`.isAssignedTo(operator) => true
+    case _ => false
+  }
+
+  private def managersOrTeamMembersOnly(operator: Operator): Boolean = operator.role match {
+    case Role.CLASSIFICATION_MANAGER | Role.CLASSIFICATION_OFFICER => true
+    case _ => false
+  }
+
+  private def managersOnly(operator: Operator): Boolean = operator.role == Role.CLASSIFICATION_MANAGER
+
+  private def adminOnly(operator: Operator): Boolean = false
```

Comment:
This one still to be finalised?

Reply:
We don't have a role for it yet, so currently nobody has this

Reply:
Arguably those roles could be deleted I guess

Reply:
Personally I think we shouldn't leave stubs like this in if it's not implemented as it'll be confusing for a new team - if they have a task that involves adding a new type of role and supporting it, they'd add it all in at the time. What benefit do we give providing something unused at this stage?

Reply:
Agreed, I'll remove